File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> curr= Arrays.asList(1);
        ans.add(curr);
        for(int i=1;i<numRows;i++){
            curr= new ArrayList<>(ans.get(i-1));
            curr.add(1);
            for(int j=curr.size()-2;j>0;j--){
                curr.set(j, curr.get(j)+curr.get(j-1));
            }
            ans.add(curr);
        }
        return ans;
    }
}

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> pascalTriangle(n);
        vector<int> row;
        for(int i=1;i<=n;i++){
            row.push_back(1);
            for(int j=i-2;j>0;j--){
                row[j]+=row[j-1];
            }
            pascalTriangle[i-1]= row;
        }
        return pascalTriangle;
    }
};